erpnext/accounts/doctype/purchase_invoice/purchase_invoice.py

```diff
@@ -435,7 +435,7 @@ def validate_supplier_invoice(self):
 		if self.bill_no:
 			if cint(frappe.db.get_single_value("Accounts Settings", "check_supplier_invoice_uniqueness")):
 				pi = frappe.db.exists("Purchase Invoice", {"bill_no": self.bill_no,
-					"fiscal_year": self.fiscal_year, "name": ("!=", self.name), "docstatus": ("<", 2)})
```

Comment:
- [ ] This will cause havoc! We need to filter using `posting_date between year_start_date and year_end_date`